server/vulnerabilities/oval/analyzer.go

```diff
@@ -93,12 +92,21 @@ func Analyze(
 		return nil, err
 	}
 
-	inserted := make([]fleet.SoftwareVulnerability, 0, len(toInsertSet))
+	var inserted []fleet.SoftwareVulnerability
+	if collectVulns {
+		inserted = make([]fleet.SoftwareVulnerability, 0, len(toInsertSet))
+	}
+
 	err = batchProcess(toInsertSet, func(v []fleet.SoftwareVulnerability) error {
-		if _, err := ds.InsertVulnerabilities(ctx, v, fleet.OVAL); err != nil {
+		n, err := ds.InsertVulnerabilities(ctx, v, fleet.OVAL)
+		if err != nil {
 			return err
 		}
-		inserted = append(inserted, v...)
+
+		if collectVulns && n > 0 {
```

Comment:
What is the `n > 0` check for?

Reply:
`InsertVulnerabilities` returns the number of rows inserted (`n`), if we are collecting results (`collectVulns == True`), we only want to return newly inserted rows.

Reply:
Hmm, wouldn't this append all rows, not just the inserted ones? I am asking because the previous version of the code did not check `n`. I wasn't sure if this changes the behavior, or if it fixed a bug in the previous version.

Reply:
A delta is computed between detected vulnerabilities and what already exists on the database, so in theory, `toInsertSet` only contains 'new' vulnerabilities.

Reply:
The part that I missed was `filterRecentVulns` in `cron.go`. It's not immediately obvious, but if `inserted` has a length of 0 when returned here, then no vulnerabilities will be considered recent from oval. If `inserted` has a length > 0, then we know at least 1 vulnerability was recent.